### PR TITLE
Implement other runtime invokers 

### PR DESCRIPTION
**Pull request by bboure**

### CODE REVIEW

**Comment by theoribeiro** [comment]
Resolved conflicts @bboure, sorry it took me a while to see there were merge conflicts on this PR.

File: src/getAppSyncConfig.js [hunk 1]
```diff
@@ -229,3 +214,37 @@ export default function getAppSyncConfig(context, appSyncConfig) {
...
+
+  // Save original write so we can restore later
+  process.stdout._orig_write = process.stdout.write;
+  process.stdout.write = (data) => {
+    output += data;
+    process.stdout._orig_write(data);
+  };
+
+  // Run serverless invoke local
+  await context.serverless.pluginManager.run(["invoke", "local"]);
```
Comment (bboure) [inline src/getAppSyncConfig.js:233]: @theoribeiro  what does this do internally? 
I have a concern here. Let's say I am using [the webpack](https://github.com/serverless-heaven/serverless-webpack)
Currently, webpack live re-builds my lambdas as I save files. When Lambdas are invoked, the transpiled copy is run. (Thanks to the `location` config)

Invoking `sls invoke local` manually in the console, transpiles the function before running it. 

So, what would happened here? When will lambdas be built? Would it happen twice? (as I save, and when running?)

**Review by bboure — COMMENTED** [review]
Thanks @theoribeiro 
No problem, I hadn't had time to look at it or test it either. 

Just had a quick look.
I am not sure if using the underlying sls invoke local is the best approach here, though I am not saying it's wrong ether. 

My main concern here is compatibility. Dropping part of aws amplify-cli in favor of using sls directly does not sound bad per say, but it might also break things and prevent future features from amplify.

There are many things that come to my mind that I am uncertain if they will keep working as they do now:
- Webpack (As I mentioned in a previous comment)
- PIPELINE resolvers (I think they should work, but not sure)
- Speed (sls invoke is quite slow in my experience, but that could be because of webpack)

Thanks again

**Comment by theoribeiro** [comment]
> Thanks @theoribeiro
> No problem, I hadn't had time to look at it or test it either.
> 
> Just had a quick look.
> I am not sure if using the underlying sls invoke local is the best approach here, though I am not saying it's wrong ether.
> 
> My main concern here is compatibility. Dropping part of aws amplify-cli in favor of using sls directly does not sound bad per say, but it might also break things and prevent future features from amplify.
> 
> There are many things that come to my mind that I am uncertain if they will keep working as they do now:
> 

> Thanks again

Hey @bboure, 

Using aws amplify-cli is definitely an option but then we would have to request some upstream changes since the way it currently calls Python functions only works for the exact same project structure aws amplify expects. This removes the flexibility given by Serverless Framework and, in my opinion, much of the benefits are lost. In my experience, PRs to Amplify usually take a long time to be analyzed and merged hence going straight to SLS would be a faster way to achieve this.

AFAIK, we're relying pretty much on what `sls offline` does behind the scenes. It does the same as `sls invoke local` for every HTTP call it receives. Let me address some of your concerns: 

> * Webpack (As I mentioned in a previous comment)

I haven't tested with Webpack yet as I do most of my development in Python but I can try to give it a shot soon.

> * PIPELINE resolvers (I think they should work, but not sure)

Same as above I think they should work but will do some testing.

> * Speed (sls invoke is quite slow in my experience, but that could be because of webpack)

What takes long to run sls invoke is usually loading all the plugins and their lifetime cycles. The proposed solution in this PR addresses this not by running sls invoke from the command line but pretending to be the command line and only invoking the plugin, therefore skipping the costly process of initializing sls. Again, I don't know if that would work for webpack.

Let me know your thoughts and thanks for this great plugin, it's been pretty useful for me!

**Comment by bboure** [comment]
Thank you @theoribeiro and sorry for not taking care of this sooner. This is a tricky one.

Have you seen @seanvm's solution in #74 ?
This is a different approach to the same problem and your feedback would be more than welcome.

**Pull request closed without merging**